src/domain/organisation/contextProviders/useStoredOrganisationState.tsx

```diff
@@ -0,0 +1,95 @@
+import { useOidcClient } from 'hds-react';
+import React from 'react';
+
+import { OrganisationContextType } from './OrganisationContext';
+import OrganisationStoragePersistor from './OrganisationStoragePersistor';
+import { useMyProfileQuery } from '../../../generated/graphql';
+import { getSelectedOrganisation } from '../../myProfile/utils';
+
+/**
+ * Fetches the user's profile data, specifically their organisations, when the user is authenticated.
+ *
+ * @returns The result of the `useMyProfileQuery` hook.
+ */
+function useProfileOrganisationsQuery() {
+  const { isAuthenticated } = useOidcClient();
+  const isLoggedIn = !isAuthenticated(); // Skip when NOT logged in
+  return useMyProfileQuery({ skip: isLoggedIn });
```

Comment:
It looked like the comment and code variable name were out-of-sync. How about this?
```suggestion
  const isNotLoggedIn = !isAuthenticated();
  return useMyProfileQuery({ skip: isNotLoggedIn });
```